File: JudgeRouteCircle5/15_04_2018.java
import java.lang.* ;
import java.util.*;

class Solution {
    public boolean judgeCircle(String moves) {
        int x = 0;
        int y = 0;
        for (char ch : moves.toCharArray()) {
            if (ch == 'U') y++;
            else if (ch == 'D') y--;
            else if (ch == 'R') x++;
            else if (ch == 'L') x--;
        }
        return x == 0 && y == 0;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        String a = new String();
        String b = new String();
        a="RLUD";
        b="LL";
        if(s.judgeCircle(a) == true) 
        {
            System.out.println("YES");
        } else 
        {
        System.out.println("NO");
        }
       if(s.judgeCircle(b) == true) 
        {
            System.out.println("YES");
        } else 
        {
        System.out.println("NO");
        }
    }
}

File: JudgeRouteCircle5/15_04_2018.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

bool judgeCircle(char* moves) {
    char *str=moves;
    int up=0,left=0;
    int leng=strlen(str);
    for(int i=0;i<leng;i++){
        switch (*(str+i)){
            case 'U':
                    up++;
                    break;
            case 'D':
                    up--;
                    break;
            case 'L':
                    left++;
                    break;
            case 'R':
                    left--;
                    break;
        }
    }
    if( up==0 && left==0)
        return true;
    return false;
}

void affiche(bool a){
    if (a == true)
    {
        printf("YES, you return your begin point\n");
    } else 
    {
        printf("NO, you lost your way or your life\n");
    }
}

int main()
{
    char *m=malloc(sizeof(char));
    char *m2=malloc(sizeof(char));
    printf("Code your way that you wanna go ! \n");
    scanf("%s",m);
    affiche(judgeCircle(m));
    free(m);
    printf("Code again your way that you wanna go ! \n");
    scanf("%s",m2);
    affiche(judgeCircle(m2));
    free(m2);
    return 0;
}